test/e2e/adminapi.go

```diff
@@ -64,7 +66,7 @@ func adminRequest(ctx context.Context, method, path string, params url.Values, i
 		return nil, err
 	}
 	defer func() {
-		_, _ = resp.Body.Read(nil)
+		adminResponseBody, _ = ioutil.ReadAll(resp.Body)
```

Comment:
Continuing [this thread](https://github.com/Azure/ARO-RP/pull/1885#discussion_r831149989):

I now see that we read it in `test/e2e/adminapi_cluster_getlogs.go` (gopls didn't find the refrence for some reason).

But using a global variable is usually a bad idea. Can we do it without a global variable? I think this has a great potintioal for a flake as it shares state between tests. It is not a problem right now as we only have one test relying on `adminResponseBody`, but it might become a problem if/when we add more.

I see that `adminRequest` already has `out interface{}` and the code which decodes the response into a struct in case when content-type is `application/json`:

https://github.com/Azure/ARO-RP/blob/2863a4735c9e278cf72dcc84a49cb538df622109/test/e2e/adminapi.go#L71-L73

I think we might want to extend it to other content-types so we can do validation by providing `out` instead of using a global var.

Reply:
Yup .. again my bad .. corrected it.